test/test_ITF1788/libieeep1788_overlap.jl

```diff
@@ -0,0 +1,162 @@
+@testset "minimal_overlap_test" begin
+
+    @test_broken MISSING_overlap(emptyinterval(), emptyinterval()) === bothEmpty
```

Comment:
For https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/504, I'm coding the `overlap` function; my idea is to return a `Symbol`, such as `:both_empty` or `:before`. Does it make sense to change this file accordingly? (Sorry for the late request.)

Reply:
See https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/504/commits/bb4d665b5f45643948b4da9f349b9eb81fb413b8 in #504.

Reply:
good point. I can change it, an alternative would be to implement the overlap states with `@enum`. Not sure which one is better 🤔 

Btw, since the overlap is fairly independent by the rest of the work, I think it would make more sense to have it as a stand-alone PR to master, maybe continuing from #266 .

Reply:
I didn't know about #266, and it LGTM. There are minor differences with respect to https://github.com/JuliaIntervals/IntervalArithmetic.jl/commit/bb4d665b5f45643948b4da9f349b9eb81fb413b8 though, the major being that it returns `String`s instead of a `Symbol`; I would favor the second one. (Your proposal with `@enum` may be even better, for performance if `overlap` is used in the actual code.)

If you think #266 is worth merging, go ahead; #271 (and $504) will have to incorporate it.

Reply:
I'll have a look at #266 tomorrow or on Tuesday. If you think it's a nice idea I could try to build on top of that with `@enum` (if you haven't done it already)